conn.go

```diff
@@ -1107,13 +1107,13 @@ func (c *conn) ResetSession(ctx context.Context) error {
 		// Just release
 		_ = c.closeNotLocking()
 	}
-	dpiConn, err := c.drv.acquireConn(pool, P)
+	dpiConn, newConnectionCreated, err := c.drv.acquireConn(pool, P)
 	c.mu.Unlock()
 	if err != nil {
 		return fmt.Errorf("%v: %w", err, driver.ErrBadConn)
 	}
 	c.dpiConn = dpiConn
-
+	ctx = context.WithValue(ctx, dsn.OnInitNewConnectionKey, newConnectionCreated)
```

Comment:
Why do we put this information into the context?

Reply:
See comment bellow. I don't want to change OnInit method and cause compile errors in other projects, the I use Context to provide info.

Reply:
Why would OnInit need this info?
It should be called iff it is a new connection.

Reply:
I reworked according your requirements. Parameter = OnInitNewCon if true , then OnInit is called only by the new connection.
Please check.